python311Packages.responses: 0.23.3 -> 0.24.0

fabaff opened the pull request:
Diff: https://github.com/getsentry/responses/compare/refs/tags/0.23.3...0.24.0

Changelog: https://github.com/getsentry/responses/blob/0.24.0/CHANGES

## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

natsukium commented on pkgs/development/python-modules/responses/default.nix:
```diff
@@ -17,7 +17,7 @@
+  version = "0.24.0";
   format = "setuptools";
 
   disabled = pythonOlder "3.7";
```
```suggestion
  disabled = pythonOlder "3.8";
```
https://github.com/getsentry/responses/blob/5579f8ecbe4bc59e04bc334f98c8a68b93c57cb3/setup.py#L79

fabaff replied:
Thanks

fabaff commented:
@ofborg build python311Packages.responses python311Packages.google-auth python311Packages.detect-secrets
 python311Packages.pygitguardian

mweinelt commented on pkgs/development/python-modules/responses/default.nix:
```diff
@@ -11,24 +11,23 @@
   pname = "responses";
-  version = "0.23.3";
+  version = "0.24.0";
   format = "setuptools";
```
Enable the PEP517 build.

fabaff committed:
python311Packages.responses: 0.23.3 -> 0.24.1

Diff: https://github.com/getsentry/responses/compare/refs/tags/0.23.3...0.24.1

Changelog: https://github.com/getsentry/responses/blob/0.24.1/CHANGES
